server/storage/players.lua

```diff
@@ -382,6 +410,124 @@ local function cleanPlayerGroups()
     lib.print.info('Removed invalid groups from player_groups table')
 end
 
+---@param citizenid string
+---@param key string | string[]
+---@param value any
+local function addPlayerDataUpdate(citizenid, key, value)
+    local hasSubKeys = type(key) == 'table'
+
+    if hasSubKeys then
+        key[1] = otherNamedPlayerFields[key[1]] or key[1]
+    else
+        key = otherNamedPlayerFields[key] or key
+    end
+
+    if jsonPlayerFields[hasSubKeys and key[1] or key] == nil then
+        lib.print.error(('Tried to update player data field %s when it doesn\'t exist. Value: %s'):format(hasSubKeys and key[1] or key, value))
+        return
+    end
+
+    if hasSubKeys and not jsonPlayerFields[key[1]] then
+        lib.print.error(('Tried to update player data field %s as a json object when it isn\'t one'):format(key[1]))
+        return
+    end
+
+    value = type(value) == 'table' and json.encode(value) or value
+
+    local currentTable = isUpdating and playerDataUpdateQueue or collectedPlayerData
+    if not currentTable[citizenid] then
+        currentTable[citizenid] = {}
+    end
+
+    currentTable[citizenid][hasSubKeys and key[1] or key] = hasSubKeys and {} or value
+
+    if hasSubKeys then
+        local current = currentTable[citizenid][key[1]]
+        if #key > 2 then
+            -- We don't check the last one because otherwise we lose the table reference
+            for i = 2, #key - 1 do
+                if not current[key[i]] then
+                    current[key[i]] = {}
+                end
+
+                current = current[key[i]]
+            end
+        end
+
+        current[key[#key]] = value
+    end
```

Comment:
This could be reworked to reduce nesting

Reply:
Not if you want to keep it readable

Reply:
Your dealing with the edge case in the first if statement, inverting the first statement into a guard clause will definitely improve readability.

Reply:
Done